test/sumlog.jl

```diff
@@ -0,0 +1,25 @@
+@testset "sumlog" begin
```

Comment:
Some surprises:
```julia
julia> sumlog([1,2,-0.1,-0.2])
-3.2188758248682

julia> sumlog([1,2,NaN])
ERROR: DomainError with NaN:
Cannot be NaN or Inf.
Stacktrace:
  [1] (::Base.Math.var"#throw1#5")(x::Float64)
    @ Base.Math ./math.jl:845
  [2] exponent
    @ ./math.jl:848 [inlined]

julia> sumlog([-0.0])
ERROR: DomainError with -0.0:
Cannot be ±0.0.

julia> sum(log, -0.0)
-Inf
```

Reply:
Calling `Base.Math._exponent_finite_nonzero` works around the NaN problem (although not for BigFloats).

Adding `xj < 0 && Base.Math.throw_complex_domainerror(:log, xj)` doesn't seem to cost much speed.

Reply:
Also, note that tests right now only test Float64